fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/functions/agg/Geomean.java

```diff
@@ -0,0 +1,93 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package org.apache.doris.nereids.trees.expressions.functions.agg;
+
+import org.apache.doris.catalog.FunctionSignature;
+import org.apache.doris.nereids.exceptions.AnalysisException;
+import org.apache.doris.nereids.trees.expressions.Expression;
+import org.apache.doris.nereids.trees.expressions.functions.ExplicitlyCastableSignature;
+import org.apache.doris.nereids.trees.expressions.shape.BinaryExpression;
+import org.apache.doris.nereids.trees.expressions.visitor.ExpressionVisitor;
+import org.apache.doris.nereids.types.DataType;
+import org.apache.doris.nereids.types.DoubleType;
+
+import com.google.common.base.Preconditions;
+import com.google.common.collect.ImmutableList;
+
+import java.util.List;
+
+/**
+ * AggregateFunction 'geomean'.
+ */
+public class Geomean extends NullableAggregateFunction
+        implements BinaryExpression, ExplicitlyCastableSignature {
+
+    public static final List<FunctionSignature> SIGNATURES = ImmutableList.of(
+            FunctionSignature.ret(DoubleType.INSTANCE).args(DoubleType.INSTANCE)
+    );
+
+    /**
+     * constructor with 1 argument.
+     */
+    public Geomean(Expression arg) {
+        this(false, arg);
+    }
+
+    /**
+     * constructor with 2 arguments.
+     */
+    public Geomean(boolean distinct, Expression arg) {
+        this(distinct, false, arg);
+    }
+
+    public Geomean(boolean distinct, boolean alwaysNullable, Expression arg) {
+        super("geomean", distinct, alwaysNullable, arg);
+    }
+
+    @Override
+    public void checkLegalityBeforeTypeCoercion() throws AnalysisException {
+        DataType dataType = getArgument(0).getDataType();
+        if ((!dataType.isNumericType() && !dataType.isNullType()) || dataType.isOnlyMetricType()) {
```

Comment:
why add this restirction? do not accept string column that all its value could be cast to double?